### PR TITLE
add jwt authentication

### PR DESCRIPTION
**Pull request by TonyH277**

### CODE REVIEW

File: src/main/java/mate/academy/bookshop/controller/AuthenticationController.java [hunk 1]
```diff
@@ -25,4 +27,9 @@ public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto r
             throws RegistrationException {
         return authenticationService.register(request);
     }
+
+    @PostMapping("login")
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/controller/AuthenticationController.java]:
```suggestion
    @PostMapping("/login")
```

Comment (d1sam) [reply src/main/java/mate/academy/bookshop/controller/AuthenticationController.java]: what about swagger? response status?

File: src/main/java/mate/academy/bookshop/dto/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+
+public record UserLoginRequestDto(
+        @Email
+        @NotBlank(message = "can not be blank")
+        String email,
+
+        @NotBlank(message = "can not be blank")
+        String password) {
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/dto/UserLoginRequestDto.java:15]: let's validate the size also

File: src/main/java/mate/academy/bookshop/dto/UserLoginRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package mate.academy.bookshop.dto;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+
+public record UserLoginRequestDto(
+        @Email
+        @NotBlank(message = "can not be blank")
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/dto/UserLoginRequestDto.java]: let's validate the size also

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -6,3 +6,6 @@ spring.datasource.password=root
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
+
+jwt.expiration=300000
+jwt.secret=dpas0-dik-a90jdf09gh903r032opnpfsad{}320rjnsf/>OPJ/f
```
Comment (d1sam) [inline src/main/resources/application.properties]: let's provide these values from env properties

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but there are some minor changes, that should be done according my comments.

File: src/main/java/mate/academy/bookshop/config/security/SecurityConfig.java [hunk 5]
```diff
@@ -43,7 +58,20 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                                 .authenticated()
                 )
                 .httpBasic(withDefaults())
+                .addFilterBefore(jwtAuthenticationFilter,
+                        UsernamePasswordAuthenticationFilter.class)
                 .userDetailsService(userDetailsService)
                 .build();
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookshop/config/security/SecurityConfig.java:64]: For JWT security it's also good to make a session management stateless.

File: src/main/java/mate/academy/bookshop/controller/AuthenticationController.java [hunk 6]
```diff
@@ -13,16 +17,26 @@
...
     @PostMapping("/register")
     @ResponseStatus(HttpStatus.CREATED)
     public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto request)
             throws RegistrationException {
         return authenticationService.register(request);
     }
+
+    @Operation(summary = "Login user", description = "Endpoint for login user")
+    @PostMapping("/login")
+    @ResponseStatus(HttpStatus.OK)
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookshop/controller/AuthenticationController.java:38]: 200 is a default status, it's not necessary to specify it.

File: src/main/java/mate/academy/bookshop/util/JwtUtil.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    public boolean isValid(String token) {
+        try {
+            Jws<Claims> claimsJws = Jwts.parser()
+                    .verifyWith(secret)
+                    .build()
+                    .parseSignedClaims(token);
+            return !claimsJws.getPayload().getExpiration().before(new Date());
+        } catch (JwtException | IllegalArgumentException e) {
+            throw new JwtException("Expired or invalid JWT token");
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookshop/util/JwtUtil.java]:
```suggestion
            return false;
```

**Review by vovasalyha — APPROVED** [review]
Approving, but left few comments for consideration.

**Pull request merged**